### DISCUSSION TITLE
Pop Shell Future

### DISCUSSION THREAD

**Discussion by Malix-Labs** [pop-os/shell#1728] [General]
Hello !

Since Pop_OS! [default desktop environment](https://support.system76.com/articles/desktop-environment/#) seems to plan switching from GNOME to [COSMIC](https://system76.com/cosmic) eventually,

- First of all, I can't seem to find any official announcement about Pop_OS! [default desktop environment](https://support.system76.com/articles/desktop-environment/#) eventually plan switching from GNOME to [COSMIC](https://system76.com/cosmic), is that true ?
- What is the planned future of Pop Shell ?
- Is Pop Shell still officially maintained ?
	- If yes, until when ?
- Is the [Pop Shell's proposal about upstreaming it to GNOME](https://github.com/pop-os/shell#the-proposal) still active ?

**Comment by jacobgkau** [comment]
> First of all, I can't seem to find any official announcement about Pop_OS! [default desktop environment](https://support.system76.com/articles/desktop-environment/#) eventually plan switching from GNOME to [COSMIC](https://system76.com/cosmic), is that true ?

Yes, COSMIC Epoch will be the default in 24.04 LTS. The first alpha has been released already.

> Is Pop Shell still officially maintained ? If yes, until when ?

It is still maintained as part of Pop!_OS 22.04 LTS. It is not being maintained for 24.04 LTS, and support for the GNOME 42 (which 22.04 LTS shipped with) will end when 22.04 LTS becomes end-of-life, which is April 2027 to match Ubuntu.

> Is the [Pop Shell's proposal about upstreaming it to GNOME](https://github.com/pop-os/shell#the-proposal) still active ?

No. As far as I'm aware, the upstream GNOME team is not interested in implementing something as complex as Pop!_Shell. They are, of course, welcome to use our code under the open-source license, but we've always had to put in a lot of work to keep up with the latest GNOME changes, and Pop!_Shell already has some problems on newer versions of GNOME that have not shipped in Pop!_OS.

> What is the planned future of Pop Shell ?

System76 is not currently putting development resources into adapting Pop!_Shell to newer GNOME versions. For the time being, we're happy to accept contributions from anyone who wants to help keep it working after GNOME updates, with minimal quality assurance barriers for GNOME versions that do not impact Pop!_OS. There's currently no plan for Pop!_Shell beyond that.

Our focus has moved to COSMIC Epoch, where tiling is handled by the COSMIC compositor, [cosmic-comp](https://github.com/pop-os/cosmic-comp). You're welcome to give it a try, either now or once it has a stable release.

You could also try other GNOME Shell extensions, such as [Forge](https://github.com/forge-ext/forge), which was inspired by Pop!_Shell and i3, and was primarily developed by a long-time community contributor to Pop!_Shell-- although I see its readme was updated just this morning to state that it's unmaintained. Other similar alternatives may exist.

You are also welcome to help keep Pop!_Shell up-to-date yourself, or to fork it into your own project if you'd like.

**Reply by Malix-Labs** [reply]
Hello and thanks for your reply.

It's very late so I'm only replying for this part for now:

I came here to ask for this question after I posted a reddit post announcing that Forge has been archived (it was unarchived to permit a discussion thread to find a maintainer); however, it is unmaintained.

If I understand correctly, it means that there will soon be no longer any proper tiling window manager for GNOME.

**Reply by jmmaranan** [reply]
Hey @jacobgkau! Thank you for mentioning Forge. And congratulations to System76 on COSMIC! I might try it out soon. 🎉

Hey @Malix-off , I still use my extension, at least stable for my daily use. I think it will still be usable for a very long time unless the upstream GNOME team changes their API. With that said I can keep up with the new versions of gnome shell so it gets released to EGO for those who do not code or can't build themselves. But any breaking changes, I might not have the time to make updates. For what it's worth, I think the biggest change from GNOME upstream was converting to import/exports module.

**Comment by aszmal** [comment]
Hello, when will pop os alpha 2? in 1 there is no sleep mode and user mode and assistance for shortsightedness

**Reply by mmstick** [reply]
You can install the latest updates through the COSMIC Store.

**Comment by aszmal** [comment]
I know I updated through the terminal and the store but I had to remove the pop from the laptop because there were no functions or there is a power supply etc

**Reply by jacobgkau** [reply]
This is a discussion post about the GNOME Pop!_Shell plugin. This repo does not concern the alpha or unrelated issues with it. Please make those posts in the [cosmic-epoch repo](https://github.com/pop-os/cosmic-epoch), or discuss them in the [Pop!_OS Mattermost chat](https://chat.pop-os.org).
